BusinessLogic/Services/TestResultService.cs

```diff
@@ -26,35 +26,40 @@ public TestResultProjection GetTestResult(TestProgressData testProgressData)
         return CreateTestResultProjection(testResult);
     }
 
-    private bool CalculateHearingLoss(List<ToneAudiometryAnswer> answers)
+    private bool CalculateHearingLoss(List<ToneAudiometryQuestionResult> answers)
     {
         bool hasHearingLoss = false;
 
-        foreach (ToneAudiometryAnswer answer in answers)
+        foreach (ToneAudiometryQuestionResult answer in answers)
         {
             if (hasHearingLoss == true) break;
 
             FrequencyMap frequencyMap = FrequencyMapping.Frequencies.First(x => x.Frequency == answer.Frequency);
             int min = frequencyMap.HearingLoss.Min;
             int max = frequencyMap.HearingLoss.Max;
 
-            if (min <= answer.LowestLimitDecibels && max >= answer.LowestLimitDecibels) hasHearingLoss = true;
-        } 
-        
+            if (min <= answer.LowestDecibels && max >= answer.LowestDecibels) hasHearingLoss = true;
+        }
+
         return hasHearingLoss;
     }
 
     private TestResult CreateTestResult(bool hasHearingLoss, TestProgressData testProgressData)
     {
+        Guid testResultId = Guid.NewGuid();
+
+        testProgressData.ToneAudiometryQuestionResults.ForEach(item => item.TestResultId = testResultId);
+        testProgressData.TextQuestionResults.ForEach(item => item.TestResultId = testResultId);
+
         return new TestResult()
         {
-            Id = Guid.NewGuid(),
+            Id = testResultId,
             TargetAudience = testProgressData.Test.TargetAudience.Label,
             TestDateTime = DateTime.Now,
             Duration = 0, // TODO Temporary until timer is implemented
```

Comment:
haal de comment maar weg of voeg timer toe?

Reply:
ik zou zulke comments wel bewaren met het argument dat het handig voor  'future development'